### PR TITLE
[EDE-641] Replace nose with pytest.

### PR DESCRIPTION
**Pull request by tehreem-sadat**
**Description:**
Replace nose with pytest.

### CODE REVIEW

File: submissions/tests/test_api.py [hunk 1]
```diff
@@ -169,11 +169,11 @@ def test_get_submission(self):
             api.get_submission("deadbeef-1234-5678-9100-1234deadbeef")
 
     @mock.patch.object(Submission.objects, 'get')
-    @raises(api.SubmissionInternalError)
     def test_get_submission_deep_error(self, mock_get):
         # Test deep explosions are wrapped
-        mock_get.side_effect = DatabaseError("Kaboom!")
-        api.get_submission("000000000000000")
+        with raises(api.SubmissionInternalError):
```
Comment (ziafazal) [inline submissions/tests/test_api.py]: we should avoid dependency on pytest instead use `TestCase`'s `assertRaises` method 
`with self.assertRaises(api.SubmissionInternalError):`

**Review by ziafazal — CHANGES_REQUESTED** [review]
@tehreem-sadat need to change the way we assert exceptions.

File: submissions/tests/test_api.py (hunk 1 above)
Comment (tehreem-sadat) [reply submissions/tests/test_api.py]: @ziafazal changes have been made. please review it again.

**Review by ziafazal — APPROVED** [review]
👍

**Comment by tehreem-sadat** [comment]
Upstream PR: https://github.com/edx/edx-submissions/pull/124 has been merged.

**Pull request merged**